server_src/lib/serverCommands.js

```diff
@@ -16,28 +22,75 @@ class SystemCommands {
       if (isNaN(parseInt(stdout))) {
         throw "No process number returned";
       }
-      logger.info("stdout:", stdout);
       checkForNamedService = true;
       // If the process had a integer pid succeeded then we are ok to restart, errors are caught by the try, catch
       exec("pm2 restart OctoFarm");
     } catch (err) {
-      logger.error(err);
+      loggerServer.error(err);
     }
     return checkForNamedService;
   }
+  // This will need changing when .deb / installation script becomes a thing. It's built to deal with the current implementation.
+  static async updateOctoFarm() {
+    loggerServer.info("Update OctoFarm server requests");
+    let haveWeSuccessfullyUpdatedOctoFarm = false;
+    let statusTypeForUser = null;
+    let messageForUser = null;
+    try {
+      // Check to see if current dir contains a git folder... hard fail otherwise.
+      let doWeHaveAGitFolder = await isGitSync("./");
+      if (!doWeHaveAGitFolder) {
+        let notGitRepoError = "Not a git repository, manual update required...";
+        messageForUser = notGitRepoError;
+        statusTypeForUser = "error";
+        throw notGitRepoError;
+      }
+
+      // We are inside a folder which was created with git... okay to continue the pull
+      const { stdout, stderr } = await exec("git pull");
+      // Throw on any error, nobody should be editing their local github folder after this release is made. If they do, they need to manually update.
+      if (stderr) {
+        let manualChangesError =
+          "Manual changes to local OctoFarm folder detected, manual update required...";
+        messageForUser = manualChangesError;
+        statusTypeForUser = "error";
+        throw stderr;
+      }
+      // Catch if user somehow circumvents the client button checks in place.
+      if (stdout.includes("Already up-to-date")) {
+        let alreadyUpToDateError =
+          "Already up to date! Ignoring update request...";
+        statusTypeForUser = "warning";
+        messageForUser = alreadyUpToDateError;
+        throw alreadyUpToDateError;
+      } else {
+        statusTypeForUser = "success";
+        messageForUser =
+          "Update command has run successfully, you need to restart the OctoFarm service to activate these changes...";
+      }
+      haveWeSuccessfullyUpdatedOctoFarm = true;
+    } catch (err) {
+      loggerServer.error("OctoFarm Update Error: ", err);
+    }
+    return {
+      haveWeSuccessfullyUpdatedOctoFarm,
+      statusTypeForUser,
+      messageForUser,
+    };
+  }
 }
 
 class Script {
```

Comment:
- multiple classes in 1 file
- what do scripts have to do with system file-IO and update commands?